web/src/components/organisms/Asset/Asset/hooks.ts

```diff
@@ -1,31 +1,37 @@
-import { useCallback, useEffect, useState } from "react";
+import { useCallback, useEffect, useMemo, useState } from "react";
 
-import { Asset } from "@reearth-cms/components/molecules/Asset/asset.type";
+import { Asset, PreviewType } from "@reearth-cms/components/molecules/Asset/asset.type";
 import { viewerRef } from "@reearth-cms/components/molecules/Asset/Asset/AssetBody/index";
 import {
-  PreviewType,
+  Asset as GQLAsset,
+  PreviewType as GQLPreviewType,
   useGetAssetQuery,
   useUpdateAssetMutation,
 } from "@reearth-cms/gql/graphql-client-api";
 
+import { convertAsset } from "../convertAsset";
+
 export default (assetId?: string) => {
-  const [asset, setAsset] = useState<Asset>({} as Asset);
-  const [selectedPreviewType, setSelectedPreviewType] = useState<PreviewType>(PreviewType.Image);
+  const [selectedPreviewType, setSelectedPreviewType] = useState<PreviewType>("IMAGE");
   const [isModalVisible, setIsModalVisible] = useState<boolean>(false);
 
-  const { data, loading } = useGetAssetQuery({
+  const { data: rawAsset, loading } = useGetAssetQuery({
     variables: {
       assetId: assetId ?? "",
     },
   });
 
+  const asset: Asset | undefined = useMemo(() => {
+    return convertAsset(rawAsset?.asset as GQLAsset);
+  }, [rawAsset]);
+
```

Comment:
If you want to display ONLY the assets that are available, and don't need to/want to display the "Asset not found" fallback, do like below and then you don't have to type like `assets?: Assets | undefined[]`. 
If you go this way, please test this and then update all types you just updated to not allow for undefined inside the assets array!

```suggestion
  const asset: Asset | undefined = useMemo(() => {
    return convertAsset(rawAsset?.asset as GQLAsset).filter((a): a is Asset => !!a);
  }, [rawAsset]);

```

Reply:
convertAsset function returns an `Asset` not an array of assets `Asset[]` so we can't use `.filter` here.
I think you meant that I should add `.filter` in `web/src/components/organisms/Asset/AssetList/hooks.ts` not in `web/src/components/organisms/Asset/Asset/hooks.ts`
something like this right?
```
  useEffect(() => {
    const assets =
      data?.assets.nodes
        .map(asset => asset as GQLAsset)
        .map(convertAsset)
        .filter(asset => !!asset) ?? [];
    setAssetList(assets as Asset[]);
  }, [data?.assets.nodes]);
```